File: Chapter13/Swift/main.swift
//
//  main.swift
//  EverydayDataStructures-Swift
//
//  Created by William Smith on 8/4/16.
//  Copyright © 2016 Websmiths, LLC. All rights reserved.
//

import Foundation

print("Hello, World!")

var arr: [Int] = [12, 11, 13, 5, 6, 9, 18, 110, 3, 14, 27, 12]

let algo: ArraySortingAlgorithms = ArraySortingAlgorithms()

//algo.selectionSort(values: &arr)
//algo.insertionSort(values: &arr)
//algo.bubbleSort(values: &arr)
//algo.quickSort(values: &arr, low: 0, high: arr.count - 1)
//algo.mergeSort(values: &arr, left: 0, right: arr.count - 1)
//algo.bucketSort(values: &arr, maxVal: 110)
//algo.countingSort(values: &arr) //Doesn't like 0
algo.radixSort(values: &arr)
for num in 0..<arr.count
{
    print(arr[num])
}

File: Chapter13/Java/main.java


/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

import main.java.*;

/**
 *
 * @author William Smith
 */
public class main {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {

        System.out.println("Hello, World!");
//        IntegerTypes.intDemo();
//        FloatTypes.floatDemo();
//        BooleanType.boolDemo();
//        StringType.stringDemo();
//        User u1 = new User("Will", 1);
//        User u2 = new User("Bob", 2);
//        User u3 = new User("Joe", 3);
//        User u4 = new User("Ron", 4);
//        User u5 = new User("Dorothy", 5);
//        LoggedInUserArray array = new LoggedInUserArray();
//
//        array.UserAuthenticated(u1);
//        array.UserAuthenticated(u2);
//        array.UserAuthenticated(u3);
//        array.UserAuthenticated(u4);
//
//        array.UserLoggedOut(u4);
//        array.UserAuthenticated(u5);

        int[] arr = {12, 11, 13, 5, 6, 9, 18, 110, 3, 14, 27, 8, 12, 0};

        ArraySortingAlgorithms algo = new ArraySortingAlgorithms();
//        algo.insertionSort(arr);
//        algo.selectionSort(arr);
//        algo.mergeSort(arr, 0, arr.length - 1);
//        algo.quickSort(arr, 0, arr.length - 1);
//        algo.bubbleSort(arr);
//        algo.BucketSort(arr, 110);
//        algo.CountingSort(arr);
//        algo.RadixSort(arr);

        for (int i : arr)
        {
            System.out.println(i);
        }
    }
    
}
